webhooks/temporalcluster_webhook.go

```diff
@@ -317,6 +318,46 @@ func (w *TemporalClusterWebhook) validateCluster(cluster *v1beta1.TemporalCluste
 		}
 	}
 
+	if cluster.Spec.Services != nil {
+		overrides := cluster.Spec.Services.Overrides
+		if overrides != nil && overrides.Deployment != nil && overrides.Deployment.Spec != nil && overrides.Deployment.JsonPatch != nil {
+			errs = append(errs,
+				field.Forbidden(
+					field.NewPath("spec", "services", "overrides", "deployment", "jsonPatch"),
+					fmt.Sprintf("Can't set JsonPatch when Spec is set on Deployment override"),
```

Comment:
```suggestion
					"Can't set JsonPatch when Spec is set on Deployment override",
```
No need for `Sprintf` :)